### PR TITLE
Added retries flag to install

### PR DESCRIPTION
**Pull request by SeppeSoete**
Added a flag to the install command to set the amount of retries in case fetching the latest version or installing specified version fails 
(I currently set the default to 1 retry)

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -139,8 +144,16 @@ var cmdInstall = &cli.Command{
...
+			for i := c.Int("retries"); i > 0; i-- {
+				if err == nil {
+					break
+				}
+				if i > 1 { // On the last attempt we want a failed attempt to be logged as fatal
+					stump.Error("error resolving %q: %s", vers, err)
+				} else {
+					stump.Fatal("error resolving %q: %s", vers, err)
+				}
+				latest, err = migrations.LatestDistVersion(c.Context, fetcher, "go-ipfs", stable)
```
Comment (Stebalien) [inline main.go:156]: There really isn't any point in re-trying when downloading from IPFS, only over HTTP. This flag should probably be piped into `NewHttpFetcher`.

File: main.go [hunk 2]
```diff
@@ -149,6 +162,12 @@ var cmdInstall = &cli.Command{
 
 		i := lib.NewInstall(vers, c.Bool("no-check"), c.Bool("allow-downgrade"), fetcher)
 		err := i.Run(c.Context)
+		for retries := c.Int("retries"); retries > 0; retries-- {
+			if err == nil {
+				break
+			}
+			err = i.Run(c.Context)
```
Comment (Stebalien) [inline main.go:169]: We definitely shouldn't try installing multiple times. If something goes wrong when actually _installing_, something is probably very wrong.

**Review by Stebalien — CHANGES_REQUESTED** [review]
Before continuing, what's your motivation? What issue did you observe?

Instead of a "retries" option, it might make more sense to just have the HttpFetcher retry a few times with a short backoff (internally).

File: main.go (hunk 2 above)
Comment (SeppeSoete) [reply main.go:169]: You're probably right, I'll look into that as well

**Comment by Stebalien** [comment]
> The issue I observed was that I tried to run the installer, it failed to fetch the latest version and when I just straight up retried the exact same thing it did work. My motivation was to have the "meh I'll just try it again and see if it works now" already built in so that the odds of someone thinking the installer/ipfs in general is broken due to some random error would be at least a bit lower.

Got it, makes sense. In that case, yeah, the retry logic definitely belongs in the HTTP Fetcher.

**Comment by Stebalien** [comment]
(closing this PR as the change will need to happen in https://github.com/ipfs/go-ipfs/blob/cd72589cfd41a5397bb8fc9765392bca904b596a/repo/fsrepo/migrations/httpfetcher.go)

**Pull request closed without merging**